src/main/java/io/qmbot/aoc/y2022/Day17.java

```diff
@@ -0,0 +1,249 @@
+package io.qmbot.aoc.y2022;
+
+import io.qmbot.aoc.Puzzle;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class Day17 implements Puzzle {
+    @Override
+    public String part1(String input) {
+        String[][] field = new String[24276][9];
+
+        fullField(field);
+
+        int startDirection = 0;
+        int start = field.length - 5;
+        int indexF = 0;
+
+        for (int i = 0; i < 2022; i++) {
+            for (String[] string : field) {
+                if (Arrays.asList(string).contains("7")) {
+                    start = Arrays.asList(field).indexOf(string) - 4;
+                    break;
+                }
+            }
+
+            startDirection = createFigure(indexF, start).startFalling(field, input, startDirection);
+
+            if (indexF == 4) indexF = -1;
+            indexF++;
+        }
+        int high = 0;
+        for (String[] string : field) {
+            if (Arrays.asList(string).contains("7")) {
+                high = Arrays.asList(field).indexOf(string);
+                break;
+            }
+        }
+
+        high = field.length - high - 1;
+        return String.valueOf(high);
+    }
+
+    @Override
+    public String part2(String input) {
+        int highField = 150;
+        String[][] field = new String[highField][9];
+        long g = 0;
+
+        fullField(field);
+
+        int startDirection = 0;
+        int start = field.length - 5;
+        int indexF = 0;
+
+        for (long numberF = 0; numberF < 1_000_000_000L; numberF++) {
+
+            for (int i = 0; i < field.length; i++) {
+                if (Arrays.asList(field[i]).contains("7")) {
+
+                    if (i == 30 || i == 31 || i == 32) {
+                        g = g + deleteBlocks(i, field, highField);
+                        break;
+                    }
+
+                    start = Arrays.asList(field).indexOf(field[i]) - 4;
+                    break;
+                }
+            }
+
+            startDirection = createFigure(indexF, start).startFalling(field, input, startDirection);
+
+            if (indexF == 4) indexF = -1;
+            indexF++;
+        }
+
+        int high = 0;
+        for (String[] string : field) {
+            if (Arrays.asList(string).contains("7")) {
+                high = Arrays.asList(field).indexOf(string);
+                break;
+            }
+        }
+
+        g = g + (field.length - high - 1);
+        return String.valueOf(g);
+
+    }
+
+    private Figure createFigure(int indexF, int start) {
+        Figure figure;
+
```

Comment:
You can replace 'indexF' with 'step' in method args and calculate indexF here as  
`   int indexF = step % 5;`